src/app/components/v2/modal-share/modal-share.js

```diff
@@ -8,27 +8,28 @@ import Component from './modal-share-component';
 
 export const reduxConfig = { actions, reducers, initialState };
 
-const mapStateToProps = ({ modalShare, location }) => ({
-  isOpen: modalShare.isOpen,
-  currentLocation: `https://www.half-earthproject.org/maps?${  window.location.href.split('?')[1]}`,
-  urlToCopy: modalShare.urlToCopy,
-  linkActive: modalShare.linkActive,
-  coordinates: location && location.query && location.query.coordinates,
-  orientation: location && location.query && location.query.orientation,
-  shareSocialMedia: [
-    {
-      icon: facebookIcon,
-      link: `https://www.facebook.com/sharer/sharer.php?u=${`https://www.half-earthproject.org/maps?${ 
-        window.location.href.split('?')[1]}`}`,
-      className: 'facebookIcon'
-    },
-    {
-      icon: twitterIcon,
-      link: `https://twitter.com/intent/tweet?url=${`https://www.half-earthproject.org/maps?${ 
-        window.location.href.split('?')[1]}`}`,
-      className: 'twitterIcon'
-    }
-  ]
-});
+const mapStateToProps = ({ modalShare, location }) => {
+  const shareLocation = `https://www.half-earthproject.org/maps${window.location.search}`;
```

Comment:
since you've got `location` already (`{ modalShare, location }`) passed in the params you can simplify it to: `https://www.half-earthproject.org/maps${location.search}` (got rid of the window)

Reply:
It doesn't seem to be the same exact object. When I tried that, the search thing was undefined....

Reply:
with `location && location.search` I get an `undefined` on the initial state when there aren't any query params. I tried `location.search || '' ` and that solves that issue. but it seems that `location.search` doesn't include the `?` so I'd need to inject that... I kind of feel that the `window.location.search` solves all of these on its own. So not sure if it's worth using the other one.

Reply:
I see. ok, let's leave `window.location.search`, the solution proposed was just cosmetics.